### PR TITLE
APP-147 Android Sample "Photo frame"

### PR DESCRIPTION
**Pull request by maxml**
Blocked because of http://jira.kaaproject.org/browse/APP-148 - no integration with Sandbox Builder
Code is working. This task is completed.

Please, review code.

### CODE REVIEW

**Comment by rasendubi** [comment]
Please, rebase on `develop`.

**Comment by maxml** [comment]
Rebased, add licence

File: photoframe/source/PhotoFrame/app/src/main/java/org/kaaproject/kaa/demo/photoframe/fragment/BaseFragment.java [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+        final Fragment fragment = BaseFragment.getCurrentFragment((MainActivity) getActivity());
+
+        getActivity().runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+
+                if (manager.isUserAttached()) {
+                    loadSlideshow(playAlbumEvent, fragment);
+                } else {
+                    Toast.makeText(getActivity(), R.string.logout_interaction_event, Toast.LENGTH_SHORT).show();
```
Comment (ikulikov) [inline photoframe/source/PhotoFrame/app/src/main/java/org/kaaproject/kaa/demo/photoframe/fragment/BaseFragment.java]: I think that it's impossible situation when endpoint can receive events in unattached state. Please explain this case or remove this part of code.

File: photoframe/source/app/src/main/java/org/kaaproject/kaa/demo/photoframe/fragment/BaseFragment.java [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+        getActivity().runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+
+                if (manager.isUserAttached()) {
+                    loadSlideshow(playAlbumEvent, fragment);
+                } else {
+                    // If you logout, but get this event
+                    Toast.makeText(getActivity(), R.string.logout_interaction_event, Toast.LENGTH_SHORT).show();
```
Comment (ikulikov) [inline photoframe/source/app/src/main/java/org/kaaproject/kaa/demo/photoframe/fragment/BaseFragment.java:170]: Please remove the whole if-else clause. Leave only loadSlideshow invocation. Please read  [Endpoints messaging](http://docs.kaaproject.org/display/KAA/Messaging+across+endpoints) (take attention on [Attach endpoint to user section](http://docs.kaaproject.org/display/KAA/Messaging+across+endpoints#Messagingacrossendpoints-Attachendpointtouser)) and [Attaching endpoints to users](http://docs.kaaproject.org/display/KAA/Attaching+endpoints+to+users) in order to understand how this feature works.

Comment (maxml) [inline photoframe/source/app/src/main/java/org/kaaproject/kaa/demo/photoframe/fragment/BaseFragment.java:170]: http://jira.kaaproject.org/browse/KAA-1310

**Comment by vadimol** [comment]
please, resolve merge conflicts locally and update the PR

**Pull request merged**